### PR TITLE
Enable TV Support for Reviewer

### LINKED ISSUE #1643: Google TV support
Originally reported on Google Code with ID 740

```
We should make sure AnkiDroid is usable on Google TV:
- Availability on Market
- Navigation without a touchscreen

Here are interesting tips:
http://android-developers.blogspot.com/2011/08/preview-of-google-tv-add-on-for-android.html
```

Reported by `nicolas.raoul` on 2011-08-23 01:10:44

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
Enable TV Support for the reviewer

This is much better than it was, but there's still room for improvement. I'm not sure if this takes it in the right direction as most of this feels hacky.

This is fairly hacky - we use a hidden view to handle the additional nav events to show the nav drawer/options.

~~It feels like there's an extra element, so pressing up to access the menu sometimes does not work. I haven't been able to figure out why.~~

~~We get a crash from the framework when navigating back from an option menu with submenus (flags in the reviewer, for example)~~

## Fixes
Fixes #1643 (reviewer)

## How Has This Been Tested?

Android TV emulator

## Learning

We shouldn't be using action bars and should write our own UI

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/ui/TvNavigationElement.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            openOptionsMenu(activity);
+            return true;
+        }
+        return super.dispatchUnhandledMove(focused, direction);
+    }
+
+
+    @SuppressLint("RestrictedApi")
+    private void openOptionsMenu(AppCompatActivity activity) {
+        // This occasionally glitches graphically on my emulator
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/ui/TvNavigationElement.java:97]: I couldn't repro a second time

**Comment by mikehardy** [comment]
With a request for patience for just a bit - I want to hold this until just after 2.13 launches for real so as to minimize diff between branches for just a few days. Shouldn't be long though - order of days (7ish? 5 to launch 2.13, a couple days after) not weeks

**Comment by david-allison-1** [comment]
Sounds good to me, it's definitely a 2.14 feature

**Review by mikehardy — APPROVED** [review]
LGTM, I don't see a reason to wait. I understand some of the code is self-diagnosed icky but nothing beats someone that actually tests during development so I'm certainly not going to tell you that it can or should be better if you couldn't see a way at the time!

**Comment by mikehardy** [comment]
I think I resolved conflicts correctly here but I'll wait for CI to pass to see
It took a conflict from your bugfix here https://github.com/ankidroid/Anki-Android/commit/3f26535a861119f6bfe6b006ab988bac4c43c1f3#diff-e7a9fede7a630f2b14e108a4a3e66fd0R387-R394 on PR #7120 in openDrawer/closeDrawer

**Comment by david-allison-1** [comment]
Requires a quick re-review, I'm now happy with this change.

File: AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java [hunk 2]
```diff
@@ -658,14 +683,72 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+        if (keyCode != KeyEvent.KEYCODE_DPAD_DOWN && keyCode != KeyEvent.KEYCODE_DPAD_UP) {
+            return false;
+        }
+
+        // HACK: This shouldn't be required, as the navigation should handle this.
+        if (isDrawerOpen()) {
+            return false;
+        }
+
+        // HACK: We should be performing this in the base class, or allowing the view to be focused by the keyboard.
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java:743]: Actually, now the WebView is out of the picture, this can probably go away

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java:749]: Nope 👎

**Review by mikehardy — APPROVED** [review]
Vamos!

**Comment by mikehardy** [comment]
:fireworks:

**Pull request merged**